tests/utils/test_dependency_constraints.py

```
@@ -0,0 +1,160 @@
+    assert spec.contains(version), (
+        f"release builds pin {version}, which the declared constraint excludes — the "
+        f"published wheel would be incompatible with its own metadata"
+    )
+
+
+def test_no_unlisted_tvm_ffi_references():
+    """A new site naming tvm-ffi must be classified, not silently added."""
+    try:
+        grep = subprocess.run(
+            # Both the PyPI name and the VCS URL form used by overrides.
+            ["git", "grep", "-lE", "apache[-/]tvm-ffi", "--", ".", ":!3rdparty"],
```

Ruff (0.16.1)

[error] Starting a process with a partial executable path

(S607)